test/e2e/tests/comments/inline.spec.ts

```diff
@@ -16,96 +16,173 @@ interface InlineCommentCreationTestProps {
+
+      // Now focus on the editor and ensure it's ready for selection
+      await textbox.click({force: true})
+      await page.waitForTimeout(300)
+
+      // Try multiple selection strategies to ensure robustness
+      // Strategy 1: Triple-click to select paragraph
+      await textbox.click({clickCount: 3, force: true})
+      await page.waitForTimeout(500)
+
+      // Verify selection and try alternative strategy if needed
+      const isFullTextSelected = await page.evaluate((expectedText) => {
+        const selection = window.getSelection()?.toString() || ''
+        return selection.includes(expectedText)
+      }, PTE_CONTENT_TEXT)
+
+      // If triple-click didn't work, try keyboard selection
+      if (!isFullTextSelected) {
+        // Click at the beginning of the text
+        await textbox.click({position: {x: 0, y: 10}})
+        await page.keyboard.down('Shift')
+
+        // Press End to select to end of line
+        await page.keyboard.press('End')
+        await page.keyboard.up('Shift')
+        await page.waitForTimeout(300)
+      }
+
+      // Final verification that we have the correct text selected
+      await expect(async () => {
+        const selectedText = await page.evaluate(() => window.getSelection()?.toString() || '')
+        return selectedText.includes(PTE_CONTENT_TEXT)
+      }).toPass({timeout: 5000})
+    }
+
+    await attemptTextSelection()
+
+    // Check if the comment button appears
+    try {
+      await page.getByTestId('inline-comment-button').waitFor({
+        state: 'visible',
+        timeout: 5000,
+      })
+      return true
+    } catch (error) {
+      // Comment button not visible after text selection
+      return false
+    }
+  }
+
+  // Try the setup and text selection up to 3 times
+  const maxAttempts = 3
+  let commentButtonVisible = false
+
+  const baseTimeout = 60000 * 3 // 60 seconds * 3 max attempts
+  test.setTimeout(baseTimeout)
+
+  for (let attempt = 1; attempt <= maxAttempts; attempt++) {
+    commentButtonVisible = await setupAndSelectText()
+
+    if (commentButtonVisible) {
+      // Comment button is visible, proceeding with test
+      break
+    } else if (attempt < maxAttempts) {
+      // Reloading page and trying again...
+      await page.reload()
+      await page.waitForLoadState('load', {timeout: WAIT_OPTIONS.timeout * 2})
+    }
+  }
+
+  // If we couldn't get the comment button to appear after all attempts, fail the test
+  if (!commentButtonVisible) {
+    throw new Error('Comment button did not appear after multiple attempts')
+  }
 
   // 4. Click on the floating comment button to start authoring a comment.
-  await page.waitForSelector('[data-testid="inline-comment-button"]', {
-    ...WAIT_OPTIONS,
-    state: 'visible',
-  })
-
-  const button = page.locator('[data-testid="inline-comment-button"]')
-  await expect(button).toBeVisible()
-  button.click({
+  const getCommentButton = () => page.locator('[data-testid="inline-comment-button"]')
```

Comment:
I think if we can avoid using locator and use instead page.getByTestId we should :)